### PR TITLE
update typing, list dependencies in readme

### PR DESCRIPTION
**Pull request by abargar**
Fixing some small typing errors in several files, typically where Optional needed to be added.

Also added installation dependencies to the readme.

### CODE REVIEW

File: orchestration/aws/infra.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def connect_to_queue(self, queue_name: str):
         self.queue = boto_resource.get_queue_by_name(QueueName=queue_name)
         return True  # TODO Is this needed?
 
-    def listen(self, callback: callable[[dict[str, Any]], None]):
+    def listen(self, callback: Callable[[dict[str, Any]], None]):
```
Comment (lovering810) [inline orchestration/aws/infra.py:53]: 🤦🏻‍♀️ good catch, I got so high on getting rid of Typing with generics that I went too far.

File: README.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ Handy tools for common use cases in data engineering
 
 I got tired of reinventing wheels across jobs in data engineering, so I decided to make a repo for them instead. Nothing in here is specific business logic, it's intended to be mostly higher order functions so you can plug and play.
 
+### Installation Dependencies
+1. `postgresql`
```
Comment (lovering810) [inline README.md]: Propose making these hyperlinks to install docs for each:

1. [postgresql](https://www.postgresql.org/download/)
2. [poetry](https://python-poetry.org/docs/#installation)

**Review by lovering810 — CHANGES_REQUESTED** [review]
Tiny addition to docs (links to install), otherwise bang on, thank you, good fixes.

File: README.md (hunk 2 above)
Comment (abargar) [inline README.md]:
```suggestion
1. `[postgresql](https://www.postgresql.org/download/)`
```

File: README.md [hunk 3]
```diff
@@ -5,6 +5,10 @@ Handy tools for common use cases in data engineering
...
+2. `poetry`
```
Comment (abargar) [inline README.md]:
```suggestion
2. `[poetry](https://python-poetry.org/docs/#installation)`
```

**Review by lovering810 — APPROVED** [review]
🥳

**Pull request merged**